src/slider/tp-slider-nav.ts

```diff
@@ -1,5 +1,65 @@
+/**
+ * Internal dependencies.
+ */
+import { TPSliderElement } from './tp-slider';
+
 /**
  * TP Slider Nav.
  */
 export class TPSliderNavElement extends HTMLElement {
+	/**
+	 * Properties.
+	 */
+	protected template: HTMLTemplateElement | null = null;
+	protected slider : TPSliderElement | null = null;
+
+	/**
+	 * Constructor.
+	 */
+	constructor() {
+		// Initialize parent.
+		super();
+
+		// Initialize properties.
+		this.template = this.querySelector( 'template' );
+		this.slider   = this.closest('tp-slider');
+
+		// Apply template if slider is found
+		this.setTemplate();
+
+		// Add event listener.
+		this.slider?.addEventListener( 'template-set', this.setTemplate.bind( this ) );
+	}
+
+	/**
+	 * Set the template.
+	 */
+	setTemplate(): void {
+		// Bail if no template.
+		if ( ! this.template || ! this.slider ) {
+			// Exit.
+			return;
+		}
+
+		// Total slides.
+		const totalSlides = Number( this.slider?.getAttribute( 'total' ) ?? 0 );
+
+		// Clear the navigation.
+		this.innerHTML = '';
+
+		// Append the navigation items.
+		for ( let i = 1; i <= totalSlides; i++ ) {
+			// Clone the template.
+			const navItem = this.template.content.cloneNode( true ) as HTMLTemplateElement;
+
+			// Find the button inside and set its text.
+			const button = navItem.querySelector('button');
+			if ( button ) {
+				button.textContent = `${i}`; // Add slide number in button
```

Comment:
@monikatravelopia Not sure this is a good idea. This will override any content set inside the button by the user. Maybe we can add a placeholder inside? Similar to: https://github.com/Travelopia/web-components/blob/master/src/slider/tp-slider-count.ts#L27

Maybe `$index`?